modules/dailymotionBidAdapter.js

```diff
@@ -30,7 +30,18 @@ export const spec = {
     method: 'POST',
     url: 'https://pb.dmxleo.com',
     data: {
-      bidder_request: bidderRequest,
+      bidder_request: {
+        gdprConsent: {
+          apiVersion: bidderRequest?.gdprConsent?.apiVersion || 1,
+          consentString: bidderRequest?.gdprConsent?.consentString || '',
+          // Cast boolean in any case (eg: if value is int) to ensure type
+          gdprApplies: !!bidderRequest?.gdprConsent?.gdprApplies,
+        },
+        refererInfo: {
+          page: bidderRequest?.refererInfo?.page || '',
+        },
+        uspConsent: bidderRequest?.uspConsent || '',
+      },
       config: config.getConfig('dailymotion'),
       coppa: config.getConfig('coppa'),
       request: bid,
```

Comment:
I wonder if we should do the same for `bid` considering the comment, will ask there after we add this change to our submission

Reply:
Good point, doing this also for `bid` would reduce the payload for the request by roughly half.

Reply:
Would be good to measure the before / after to tackle the payload reduction at the same time.
Maybe in the future we'll change the key names to make it even shorter if necessary

Reply:
Hello @dgirardi thanks for your feedbacks :)

I believe we've applied the necessary changes on the request part (pending internal review). I'm just wondering if it's necessary to do the same for the responses. The idea is that by having the least manipulation possible, we can troubleshoot directly on our back-end service, as it would be the only place where the response object is being built.

It would also seem that the bid validity checks are being done within `bidderFactory.js` (and maybe elsewhere) and should lead to https://github.com/dailymotion-oss/Prebid.js/blob/feat/dailymotion-adaptor/src/constants.json#L129 error in case of invalidity. In case of breaking change of the logic, this would lead to our inability to bid which we would be able to react to.

Looking at some other adaptors, it would seem that this approach is being adopted elsewhere (eg: https://github.com/dailymotion-oss/Prebid.js/blob/feat/dailymotion-adaptor/modules/ad2ictionBidAdapter.js#L48). Another common approach seems to be to check the existence of keys within the response object (eg: https://github.com/dailymotion-oss/Prebid.js/blob/feat/dailymotion-adaptor/modules/addefendBidAdapter.js#L63).

Considering that it's not likely that the responses format will change (although no guarantee), and these different points, do you think it's reasonable to move forward in this manner? Or do you think we should at least do some checks on our side?

Thanks for your time!

Reply:
If I were responsible for long term maintenance of the service I would definitely not consider it reasonable, but it's your call to make - there's no requirement to decouple, or to do it now. Mine is just a recommendation; I know that adapters like those you linked would not even show up in our radar when we're rolling out a breaking change.

Reply:
Do you have any recommendation about how we could get notified of a breaking change ?

Reply:
Decouple! :) the problem with trying to get notified is that when everything is hidden server side, we don't even know what might be breaking, we gauge impact of changes by looking at the codebase.

Reply:
Hello @dgirardi, did you have any chances to look at our changes? Do not hesitate if you have any other feedback